tools/clang/lib/Sema/SemaOverload.cpp

```diff
@@ -9491,12 +9491,18 @@ static void NoteFunctionCandidate(Sema &S, OverloadCandidate *Cand,
 
   case ovl_fail_bad_conversion: {
     unsigned I = (Cand->IgnoreObjectArgument ? 1 : 0);
-    for (unsigned N = Cand->NumConversions; I != N; ++I)
-      if (Cand->Conversions[I].isInitialized() && Cand->Conversions[I].isBad()) // HLSL Change: check in and out, check out conversions
-        return DiagnoseBadConversion(S, Cand, I, Cand->Conversions[I], OpLoc); // HLSL Change: add OpLoc
-    if (Cand->OutConversions[I].isInitialized() && Cand->OutConversions[I].isBad()) // HLSL Change: check in and out, check out conversions
-      return DiagnoseBadConversion(S, Cand, I, Cand->OutConversions[I], OpLoc); // HLSL Change: add OpLoc
-
+    for (unsigned N = Cand->NumConversions; I != N; ++I) {
+      if (Cand->Conversions[I].isInitialized() &&
+          Cand->Conversions[I]
+              .isBad()) // HLSL Change: check in and out, check out conversions
```

Comment:
As long as we are reformatting this code to span multiple line, I think it would read nicer to hoist the comment to the start of the if rather than the trailing lines.

Reply:
Done. I left the "add OpLoc" line in place, since that one is specifically about the argument.